### PR TITLE
[6.0][mediacontroller] refactor module and add new methods for playlists

### PR DESCRIPTION
**Pull request by djuszczakk**
[ACR]
https://code.sec.samsung.net/jira/browse/TWDAPI-249
http://suprem.sec.samsung.net/jira/browse/TWDAPI-245
http://suprem.sec.samsung.net/jira/browse/TWDAPI-246
http://suprem.sec.samsung.net/jira/browse/TWDAPI-247
http://suprem.sec.samsung.net/jira/browse/TWDAPI-248
http://suprem.sec.samsung.net/jira/browse/TWDAPI-239

### CODE REVIEW

File: docs/application/web/guides/multimedia/media-controller.md [hunk 1]
```diff
@@ -535,22 +518,36 @@ To manage the media controller playlists in your server application, you must le
...
+        function errorCallback(error) {
+            console.log("getAllPlaylists failed with error: " + error);
+            }
+        mcServer.playlists.getAllPlaylists(successCallback, errorCallback);
+        ```
+
+    2. or get playlist by name with method `getPlaylist()`:
+
+        ```javascript
+        /* assuming that the playlist was previously created */
```
Comment (pkosko) [inline docs/application/web/guides/multimedia/media-controller.md]: basically, we do not assume anything, proper error handling is done.

File: docs/application/web/guides/multimedia/media-controller.md [hunk 2]
```diff
@@ -604,15 +601,43 @@ To manage the media controller playlist in your application, you must handle req
...
+        function errorCallback(error) {
+            console.log("getAllPlaylists failed with error: " + error);
+            }
+        mcServerInfo.playlists.getAllPlaylists(successCallback, errorCallback);
+        ```
+
+    2. or get playlist by name with method `getPlaylist()`:
+
+        ```javascript
+        /* assuming that the playlist was previously created */
```
Comment (pkosko) [inline docs/application/web/guides/multimedia/media-controller.md]: also here, the existing of playlist with given name is not necessary, code handles error.

File: docs/application/web/guides/multimedia/media-controller.md [hunk 3]
```diff
@@ -122,25 +118,37 @@ To manage the media controller features in your application, you must learn to s
...
-  Each listener must invoke the appropriate function to perform the related update on the server and send an appropriate notification to the client. For example, the `onplaybackstaterequest` listener must use the `updatePlaybackState()` method to update the playback state.
+  Each listener must invoke the appropriate function to perform the related update on the server and send an appropriate notification to the client. For example, the `onshufflemoderequest` listener can change the playback shuffle mode of the media controller server.
 
    ```javascript
    var requestPlaybackInfoCb = {
-       onplaybackstaterequest: function(state, clientName) {
-           console.log('Request to change the playback state to: ' + state);
-           mcServer.updatePlaybackState(state);
+       onplaybackactionrequest: function(action, clientName) {
+<<<<<<< HEAD
```
Comment (mijin85cho) [inline docs/application/web/guides/multimedia/media-controller.md]: When you update this PR, please don't forget to resolve this merge conflict marker also.

Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

**Comment by mijin85cho** [comment]
Hello, @surbhig12 
In this PR, only docs/application/web/guides/multimedia/media-controller.md file is requested to be checked to SRI-B reviewer.
Thanks in advance.

File: docs/application/web/guides/multimedia/media-controller.md [hunk 4]
```diff
@@ -43,17 +43,13 @@ The main features of the Media Controller API include:
 
   You can use the media controller server to [receive the search requests](#receiving-search-request) using the `setSearchRequestListener()` method, and return the `RequestReply` object to the client who sent the request.
 
-- Setting content type for the currently playing media
+- [Setting content type](#setting-content-type-for-currently-playing-media) for the currently playing media
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 5]
```diff
@@ -43,17 +43,13 @@ The main features of the Media Controller API include:
...
-
-- Setting age rating for the currently playing media
-
-  You can [set age rating for the currently playing media](#setting-content-age-rating-for-currently-playing-media) using the updatePlaybackAgeRating() server method.
+- [Setting age rating](#setting-content-age-rating-for-currently-playing-media) for the currently playing media
 
 - Managing playlists on server side
 
-  You can [manage playlists](#managing-playlists-on-server-side) on server side by creating, saving, deleting, and getting information about playlists. You can also update the playback item from the playlists item list.
+  You can [manage playlists](#managing-playlists-on-server-side) on server side by creating, saving, deleting and getting information about playlists. You can also update the playback item from the playlists item list.
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 6]
```diff
@@ -151,28 +155,24 @@ To manage the media controller features in your application, you must learn to h
...
-   In the following example, send a request for the changing the playback state to paused using the `sendPlaybackState()` method:
+4. You can send a request from the client using the `sendPlaybackAction()`, `sendPlaybackPosition()`, `sendShuffleMode()` or `sendRepeatState()` methods of the `MediaControllerServerInfoPlaybackInfo` interface (in [mobile](../../api/latest/device_api/mobile/tizen/mediacontroller.html#MediaControllerServerInfoPlaybackInfo), [wearable](../../api/latest/device_api/wearable/tizen/mediacontroller.html#MediaControllerServerInfoPlaybackInfo), and [tv](../../api/latest/device_api/tv/tizen/mediacontroller.html#MediaControllerServerInfoPlaybackInfo) applications).
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 7]
```diff
@@ -151,28 +155,24 @@ To manage the media controller features in your application, you must learn to h
...
+   In the following example, send a request for the changing the playback shuffle mode to `true` using the `sendShuffleMode()` method:
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 8]
```diff
@@ -535,22 +527,35 @@ To manage the media controller playlists in your server application, you must le
...
+
+        ```javascript
+        var playlist = null;
+        try {
+            playlist = mcServer.playlists.getPlaylist("playlistName");
+        } catch (err) {
+            console.log(err.name + ": " + err.message);
+        }
+
+        ```
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 9]
```diff
@@ -575,17 +580,17 @@ To manage the media controller playlists in your server application, you must le
...
+   mcServer.playlists.deletePlaylist(playlist.name, deleteSuccess, deleteFailure);
    ```
 
 ## Managing Playlists on Client Side
 
 To manage the media controller playlist in your application, you must handle requests from the client to the server:
 
-1. Send a request from the client using the `sendPlaybackItem()` method of the `MediaControllerServerInfo` interface (in [mobile](../../api/latest/device_api/mobile/tizen/mediacontroller.html#MediaControllerServerInfo), [wearable](../../api/latest/device_api/wearable/tizen/mediacontroller.html#MediaControllerServerInfo), and [tv](../../api/latest/device_api/tv/tizen/mediacontroller.html#MediaControllerServerInfo) applications).
+1. Send a request from the client using the `sendPlaybackItem()` method of the `MediaControllerPlaylistsInfo` interface (in [mobile](../../api/latest/device_api/mobile/tizen/mediacontroller.html#MediaControllerPlaylistsInfo), [wearable](../../api/latest/device_api/wearable/tizen/mediacontroller.html#MediaControllerPlaylistsInfo), and [tv](../../api/latest/device_api/tv/tizen/mediacontroller.html#MediaControllerPlaylistsInfo) applications).
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 10]
```diff
@@ -604,15 +609,42 @@ To manage the media controller playlist in your application, you must handle req
...
+        ```javascript
+        var playlist = null;
+        try {
+            playlist = mcServerInfo.playlists.getPlaylist("playlistName");
+        } catch (err) {
+            console.log(err.name + ": " + err.message);
+        }
+
+        ```
+
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 11]
```diff
@@ -43,13 +43,13 @@ The main features of the Media Controller API include:
 
   You can use the media controller server to [receive the search requests](#receiving-search-request) using the `setSearchRequestListener()` method, and return the `RequestReply` object to the client who sent the request.
 
-- Setting content type for the currently playing media
+- Setting content type
 
-  You can [set content type for the currently playing media](#setting-content-type-for-currently-playing-media) using the updatePlaybackContentType() server method.
+    You can [set the content type](#setting-content-type-for-currently-playing-media) for the currently playing media
```
Comment (surbhig12) [inline docs/application/web/guides/multimedia/media-controller.md]: Line 48 and 52, please add a period at the end of the sentence.

```suggestion
You can [set the content type](#setting-content-type-for-currently-playing-media) for the currently playing media.
```

File: docs/application/web/guides/multimedia/media-controller.md [hunk 12]
```diff
@@ -43,13 +43,13 @@ The main features of the Media Controller API include:
...
 
-- Setting age rating for the currently playing media
+- Setting age rating
 
-  You can [set age rating for the currently playing media](#setting-content-age-rating-for-currently-playing-media) using the updatePlaybackAgeRating() server method.
+    You can [set the age rating](#setting-content-age-rating-for-currently-playing-media) for the currently playing media
```
Comment (surbhig12) [inline docs/application/web/guides/multimedia/media-controller.md]: Line 48 and 52, please add a period at the end of the sentence.

```suggestion
  You can [set the age rating](#setting-content-age-rating-for-currently-playing-media) for the currently playing media.
```

File: docs/application/web/guides/multimedia/media-controller.md (hunk 11 above)
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md (hunk 12 above)
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 13]
```diff
@@ -122,25 +122,33 @@ To manage the media controller features in your application, you must learn to s
 To manage the media controller features in your application, you must learn to handle requests from the client to the server:
 
 1. Create a `MediaControllerChangeRequestPlaybackInfoCallback` object (in [mobile](../../api/latest/device_api/mobile/tizen/mediacontroller.html#MediaControllerChangeRequestPlaybackInfoCallback), [wearable](../../api/latest/device_api/wearable/tizen/mediacontroller.html#MediaControllerChangeRequestPlaybackInfoCallback), and [tv](../../api/latest/device_api/tv/tizen/mediacontroller.html#MediaControllerChangeRequestPlaybackInfoCallback) applications) to define listeners for getting the requests from a client.
-  Each listener must invoke the appropriate function to perform the related update on the server and send an appropriate notification to the client. For example, the `onplaybackstaterequest` listener must use the `updatePlaybackState()` method to update the playback state.
+  Each listener must invoke the appropriate function to perform the related update on the server and send an appropriate notification to the client. For example, the `onshufflemoderequest` listener can change the playback shuffle mode of the media controller server.
```
Comment (surbhig12) [inline docs/application/web/guides/multimedia/media-controller.md]: For consistency, please remove period (.) and add a colon (:) at the end of the sentence:

```suggestion
Each listener must invoke the appropriate function to perform the related update on the server and send an appropriate notification to the client. For example, the `onshufflemoderequest` listener can change the playback shuffle mode of the media controller server:
```

Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 14]
```diff
@@ -217,33 +215,31 @@ To manage the media controller features in your application, you must learn to r
...
    ```javascript
-   mcServerInfo.removePlaybackInfoChangeListener(watcherId);
+   mcServerInfo.playback.removePlaybackInfoChangeListener(watcherId);
    ```
 
 ## Sending and Receiving Custom Commands
 
 To manage the media controller features in your application, you must learn to send custom commands:
 
-<a name="send_custom_commands"></a>
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 15]
```diff
@@ -272,8 +268,6 @@ To manage the media controller features in your application, you must learn to s
       mcServerInfo.sendCommand('myPlaylistFilter', exampleCustomCommandData, sendCommandSuccessCallback, sendCommandErrorCallback);
       ```
 
-<a name="receive_custom_commands"></a>
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 16]
```diff
@@ -301,8 +295,6 @@ To manage the media controller features in your application, you must learn to s
 Custom command enables the client application to talk to the server application.
 The communication in the opposite direction is done with the help of custom events.
 
-<a name="receive_custom_events"></a>
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 17]
```diff
@@ -333,8 +325,6 @@ The communication in the opposite direction is done with the help of custom even
       mcClient.unsetCustomEventListener();
       ```
 
-<a name="send_custom_events"></a>
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 18]
```diff
@@ -122,25 +122,33 @@ To manage the media controller features in your application, you must learn to s
...
        onshufflemoderequest: function(mode, clientName) {
            console.log('Request to change the playback shuffle mode to: ' + (mode ? 'TRUE' : 'FALSE'));
-           mcServer.updateShuffleMode(mode);
+           mcServer.playback.shuffleMode = mode;
        },
-       onrepeatmoderequest: function(mode, clientName) {
-           console.log('Request to change the playback repeat mode to: ' + (mode ? 'TRUE' : 'FALSE'));
-           mcServer.updateRepeatMode(mode);
+       onrepeatstaterequest: function(state, clientName) {
+           console.log('Request to change the playback repeat state to: ' + state));
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: First ) closes the expression started in "(mode ?"
Second closes the expression for "console.log('Request "

File: docs/application/web/guides/multimedia/media-controller.md [hunk 19]
```diff
@@ -535,22 +531,35 @@ To manage the media controller playlists in your server application, you must le
...
+            });
+        }
+        function errorCallback(error) {
+            console.log("getAllPlaylists failed with error: " + error);
+            }
+        mcServer.playlists.getAllPlaylists(successCallback, errorCallback);
+        ```
+
+    * To get the playlist information by name, use the `getPlaylist()` method:
+
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 20]
```diff
@@ -604,15 +613,42 @@ To manage the media controller playlist in your application, you must handle req
...
+            });
+        }
+        function errorCallback(error) {
+            console.log("getAllPlaylists failed with error: " + error);
+            }
+        mcServerInfo.playlists.getAllPlaylists(successCallback, errorCallback);
+        ```
+
+    * To get the playlist information by name, use the `getPlaylist()` method:
+
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

File: docs/application/web/guides/multimedia/media-controller.md [hunk 21]
```diff
@@ -473,17 +466,20 @@ different types of media items.
...
 2. Accessing content age rating on the client side:
 
     ```javascript
+    var mcClient = tizen.mediacontroller.getClient();
+    var mcServerInfo = mcClient.getLatestServerInfo();
     var userAge = 17; // App developer should retrieve actual user age from user profile.
-    var rating = mcServerInfo.playbackInfo.ageRating;
+    var rating = mcServerInfo.playback.ageRating;
     if (rating > userAge) {
         console.log("Warning: this content has age rating " + rating + "+.";
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: No, it's not. I've added missing bracket in next commit.

File: docs/application/web/guides/multimedia/media-controller.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (mijin85cho) [reply docs/application/web/guides/multimedia/media-controller.md]: I can't see the "(mode ?" in this line.

console.log('Request to change the playback repeat state to: ' + state));

[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: oh, sorry, I was looking for wrong line. It should be single, I've updated this in new commit

File: docs/application/web/guides/multimedia/media-controller.md [hunk 22]
```diff
@@ -525,7 +528,7 @@ To manage the media controller playlists in your server application, you must le
    playlist.addItem("index1", metadata);
    ```
 
-3. To save playlist, use the `savePlaylist()` method:
+3. To save playlist, use the `savePlaylist()` method wih optional callbacks:
```
[the remark this answers is not in the document]
Comment (djuszczakk) [reply docs/application/web/guides/multimedia/media-controller.md]: Done

**Pull request merged**